### DISCUSSION TITLE
Shipping custom yolo weights in ODC Docker

### DISCUSSION THREAD

**Discussion by dominique-AR** [opendatacam/opendatacam#563] [Q&A] [answered]
Hi everyone,  I'm new in this topic and I'm glad if someone can help please. I begun with ODC just a few month in order to test a custom yolo weights in a local traffic stream.   I've changed config.json and docker-compose file but there's no change, ODC still stuck on the default video stream. It's possible to change weights with custom yolov4 weights or should I  just use the pre-trained weights from AlexeyAD repository?

Thanks folks.

#config.json
  "NEURAL_NETWORK_PARAMS": {
    "yolov4": {
      "data": "data/obj.data",
      "cfg": "cfg/custom-yolov4.cfg",
      "weights": "backup/custom-yolov4.weights"

#docker-compose.yml
    volumes:
        - './config.json:/home/user/config.json'
        - './opendatacam_videos:/var/local/darknet/opendatacam_videos'
        - './yolov4.weights:/var/local/darknet/custom-yolov4.weights'
        - './obj.data:/var/local/darknet/data/obj.data'
        - './yolov4.cfg:/var/local/darknet/cfg/custom-yolov4.cfg'
        - './obj.names:/var/local/darknet/data/obj.names'

**Comment by vsaw** [comment, marked as answer]
You can use custom weights in OpenDataCam with and without Docker. The setup is a bit brittle, because you have to change so many paths, but here's how I do it

`docker-compose.yml`
```
volumes:
      - './custom.json:/var/local/opendatacam/config.json'
      - './custom.weights:/var/local/darknet/custom.weights'
      - './custom.cfg:/var/local/darknet/cfg/custom.cfg'
      - './custom.data:/var/local/darknet/cfg/custom.data'
      - './custom.names:/var/local/darknet/cfg/custom.names'
      - './:/var/local/darknet/opendatacam_videos'
```

`custom.json`
```
{
    "NEURAL_NETWORK": "custom",
    // …
    "NEURAL_NETWORK_PARAMS": {
        "custom": {
            "data": "cfg/custom.data",
            "cfg": "cfg/custom.cfg",
            "weights": "custom.weights"
        }
    }
}
```

`custom.data`
```
classes = 1
names = /var/local/darknet/cfg/custom.names
```

**Reply by dominique-AR** [reply]
@vsaw thank you, I'll try it and I'll let you know the results.

**Reply by dominique-AR** [reply]
@vsaw it's worked fine. Thanks again Boss!!
